shell/platform/darwin/ios/framework/Source/FlutterKeyboardManagerTest.mm

```diff
@@ -17,364 +15,253 @@
-             if (press == event1) {
-               key1Callback = callback;
-             } else if (press == event2) {
-               key2Callback = callback;
-             }
-           }]];
-
-  // Add both presses into the main CFRunLoop queue
-  CFRunLoopTimerRef timer0 = CFRunLoopTimerCreateWithHandler(
-      kCFAllocatorDefault, CFAbsoluteTimeGetCurrent(), 0, 0, 0, ^(CFRunLoopTimerRef timerRef) {
-        [manager handlePress:event1
-                  nextAction:^() {
-                    key1Handled = false;
-                  }];
-      });
-  CFRunLoopAddTimer(CFRunLoopGetCurrent(), timer0, kCFRunLoopCommonModes);
-  CFRunLoopTimerRef timer1 = CFRunLoopTimerCreateWithHandler(
-      kCFAllocatorDefault, CFAbsoluteTimeGetCurrent() + 1, 0, 0, 0, ^(CFRunLoopTimerRef timerRef) {
-        // key1 should be completely finished by now
-        XCTAssertFalse(key1Handled);
-        [manager handlePress:event2
-                  nextAction:^() {
-                    key2Handled = false;
-                  }];
-        // End the nested CFRunLoop
-        CFRunLoopStop(CFRunLoopGetCurrent());
-      });
-  CFRunLoopAddTimer(CFRunLoopGetCurrent(), timer1, kCFRunLoopCommonModes);
-
-  // Add the callbacks to the CFRunLoop with mode kMessageLoopCFRunLoopMode
-  // This allows them to interrupt the loop started within handlePress
-  CFRunLoopTimerRef timer2 = CFRunLoopTimerCreateWithHandler(
-      kCFAllocatorDefault, CFAbsoluteTimeGetCurrent() + 2, 0, 0, 0, ^(CFRunLoopTimerRef timerRef) {
-        // No processing should be done on key2 yet
-        XCTAssertTrue(key1Callback != nil);
-        XCTAssertTrue(key2Callback == nil);
-        key1Callback(false);
-      });
-  CFRunLoopAddTimer(CFRunLoopGetCurrent(), timer2,
-                    fml::MessageLoopDarwin::kMessageLoopCFRunLoopMode);
-  CFRunLoopTimerRef timer3 = CFRunLoopTimerCreateWithHandler(
-      kCFAllocatorDefault, CFAbsoluteTimeGetCurrent() + 3, 0, 0, 0, ^(CFRunLoopTimerRef timerRef) {
-        // Both keys should be processed by now
-        XCTAssertTrue(key1Callback != nil);
-        XCTAssertTrue(key2Callback != nil);
-        key2Callback(false);
-      });
-  CFRunLoopAddTimer(CFRunLoopGetCurrent(), timer3,
-                    fml::MessageLoopDarwin::kMessageLoopCFRunLoopMode);
-
-  // Start a nested CFRunLoop so we can wait for both presses to complete before exiting the test
-  CFRunLoopRun();
-  XCTAssertFalse(key2Handled);
-  XCTAssertFalse(key1Handled);
+  id expectationEvent1Action = [self expectationWithDescription:@"event1 action"];
+  [manager handlePress:event1
+            nextAction:^() {
+              [expectationEvent1Action fulfill];
+            }];
+
+  id expectationEvent2Action = [self expectationWithDescription:@"event2 action"];
+  [manager handlePress:event2
+            nextAction:^() {
+              [expectationEvent2Action fulfill];
+            }];
+
+  [self waitForExpectations:@[
```

Comment:
Removed the runloop checks in favor of enforced order expectations, and waiting for them spins the runloop.

Reply:
Ignore, I reverted this.